### PR TITLE
Check

### PR DESCRIPTION
**Pull request by punkleia**
Working on game in check method

### CODE REVIEW

File: app/models/game.rb [hunk 1]
```diff
@@ -60,6 +61,58 @@ def initialize_board
...
   # rubocop:enable Metrics/AbcSize
   # rubocop:enable MethodLength
+
+  # returns true if game is in check
+  def black_in_check?(x, y)
+    # identify the black kings on the board
+    King.where(is_black: true, x_position: x, y_position: y)
+    # selects white pieces and looks to see if
+    # they have an open route to the king
+    pieces.where(is_black: false).each do |piece|
```
Comment (tynahuynh) [inline app/models/game.rb]: I don't think this would check to make sure the piece is actually active correct?

File: app/models/game.rb [hunk 2]
```diff
@@ -60,6 +61,58 @@ def initialize_board
     King.create(is_black: true, x_position: 4, y_position: 7, game_id: id, user_id: user_id_black, status: 'active')
     Queen.create(is_black: true, x_position: 3, y_position: 7, game_id: id, user_id: user_id_black, status: 'active')
   end
+
   # rubocop:enable Metrics/AbcSize
   # rubocop:enable MethodLength
+
+  # returns true if game is in check
+  def black_in_check?(x, y)
+    # identify the black kings on the board
```
Comment (tynahuynh) [inline app/models/game.rb]: Just curious why we want multiple black or white kings? I don't think promotion works from pawn to king?

Comment (punkleia) [inline app/models/game.rb]: Hah!  In that case we would not want it.  I thought you could promote to anything. :P

**Comment by jacobwgillespie** [comment]
I think that the check methods could become:

```ruby
def black_in_check?
  king = pieces.active.where(type: 'King', is_black: true)
  king.in_check?
end

def white_in_check?
  king = pieces.active.where(type: 'King', is_black: false)
  king.in_check?
end
```

That way you're finding the king in the current game by color then reusing the king's `in_check?` method.  You don't even need to pass in the `x` or `y` since there's only one king per color (no pawn promotion to a king).

**Comment by punkleia** [comment]
I'm still getting No Method Errors for my check method and an additional No Method Error for my turn method.  Am I missing something in the Move and Game models telling them to look at the children models?

Also, is the difference between the two of these just readability?  

`def black_in_check?
  king = pieces.active.where(type: 'King', is_black: true)
  king.in_check?
end`

`def black_in_check?
  pieces.active.where(type: 'King', is_black: true).in_check?
end`

**Comment by punkleia** [comment]
I do not like rubocop's preferred formatting suggestion of "Integer#even?"  Do I suck it up, because that's what it "should" be, or can I override that setting somewhere?

**Comment by jacobwgillespie** [comment]
Either way - I'd probably change it to `.even?` since that's more "descriptive" but you could also find the rubocop rule name and disable it in `.rubocop.yml`

**Comment by punkleia** [comment]
Check is ready to go! I believe.
I've moved turn logic to another branch.

**Comment by jacobwgillespie** [comment]
Looks like the king and pawn tests are commented out and some logic in the king has been commented as well - was that intentional?

**Comment by punkleia** [comment]
It was intentional - it wasn't logic I wrote and they kept failing tests.  I forget to uncomment them.

**Comment by punkleia** [comment]
Fixed!

**Comment by jacobwgillespie** [comment]
This looks good to me, but I'm going to un-comment out the pawn test stuff and get it working again, as it may cause some issues with other branches if this gets merged with the commented version.  I'll push the commit here in a sec.

**Comment by jacobwgillespie** [comment]
Looks like those tests just work now, so no need for fixing anymore!  LGTM, let's merge this 🚀

**Pull request merged**